botocore/signers.py

```diff
@@ -123,6 +113,36 @@ def sign(self, operation_name, request):
                                             signature_version)
             signer.add_auth(request=request)
 
+    def choose_signer(self, operation_name, suffix=''):
+        """
+        Allow overriding the signature version via the choose-signer event.
+        A value of `botocore.UNSIGNED` means no signing will be performed.
+
+        :param operation_name: The operation to sign.
+        :param suffix: A suffix that needs to be on all valid signers. This
```

Comment:
That brings us right back to the problem of having copied code all over the place.